module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -1429,7 +1429,19 @@ protected function makeRequest($hierarchy, $params = false, $method = 'GET',
 
         // Send request and retrieve response
         $startTime = microtime(true);
-        $response = $client->setMethod($method)->send();
+        try {
+            $response = $client->setMethod($method)->send();
+        } catch (\Exception $e) {
+            $params = $method == 'GET'
+                ? $client->getRequest()->getQuery()->toString()
+                : $client->getRequest()->getPost()->toString();
+            $this->error(
+                "$method request for '$apiUrl' with params '$params' and contents '"
+                . $client->getRequest()->getContent() . "' caused exception: "
+                . $e->getMessage()
+            );
+            throw new ILSException('Problem with Sierra REST API.');
```

Comment:
Is it worth passing through the previous exception here and in the other exceptions below -- e.g.`throw new ILSException('Problem with Sierra REST API.', 0, $e);`? That might allow for a more complete backtrace in case of trouble.

Reply:
Sure does, thanks. Now changed. Might be useful to modify other drivers too.

Reply:
Thanks, and good idea: I've opened [VUFIND-1436](https://vufind.org/jira/browse/VUFIND-1436) as a reminder to update other drivers. We can make a to-do list on there as time permits.